src/librustc_resolve/lib.rs

```diff
@@ -2283,24 +2283,25 @@ impl<'a> Resolver<'a> {
                 PatKind::Ident(bmode, ref ident, ref opt_pat) => {
                     // First try to resolve the identifier as some existing
                     // entity, then fall back to a fresh binding.
-                    let resolution = if let Ok(resolution) = self.resolve_path(pat.id,
-                                &Path::from_ident(ident.span, ident.node), 0, ValueNS) {
+                    let local_def = self.resolve_identifier(ident.node, ValueNS, true);
```

Comment:
I changed from `resolve_path` to `resolve_identifier` specifically to avoid the adjustments.
For example, consider:
```rust
fn f(x: i32) {
    fn g(x: i32) {}
}
```
Here, if we call `resolve_path` on `g`'s argument `x`, it "should" resolve to the argument `f`'s argument `x`. When this resolution gets adjusted it would emit 
```
can't capture dynamic environment in a fn item
```
(since we've already pushed the "rib"s for `g`), which we clearly don't want.

I say "should" because right now, `g`'s argument `x` does not resolve to `f`'s argument `x` because of an implementation detail of the current hygiene algorithm. I'm working on a [simplification of the hygiene algorithm](https://github.com/rust-lang/rust/pull/34374#issuecomment-227625734) after which `g`'s argument `x` will resolve to `f`'s argument `x`.